spec/asyncapi.md

```diff
@@ -373,7 +372,7 @@ Field Name | Type | Description
 <a name="serverObjectProtocolVersion"></a>protocolVersion | `string` | The version of the protocol used for connection. For instance: AMQP `0.9.1`, HTTP `2.0`, Kafka `1.0.0`, etc.
 <a name="serverObjectDescription"></a>description | `string` | An optional string describing the host designated by the URL. [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.
 <a name="serverObjectVariables"></a>variables | Map[`string`, [Server Variable Object](#serverVariableObject) \| [Reference Object](#referenceObject)]] | A map between a variable name and its value.  The value is used for substitution in the server's URL template.
-<a name="serverObjectSecurity"></a>security | [[Security Requirement Object](#securityRequirementObject)] | A declaration of which security mechanisms can be used with this server. The list of values includes alternative security requirement objects that can be used. Only one of the security requirement objects need to be satisfied to authorize a connection or operation.
+<a name="serverObjectSecurity"></a>security | [[Security Scheme Object](#securitySchemeObject) \| [Reference Object](#referenceObject)] | A declaration of which security schemes can be used with this server. The list of values includes alternative [security scheme objects](#securitySchemeObject) that can be used. Only one of the security scheme objects need to be satisfied to authorize a connection or operation.
```

Comment:
Instead of having a weird syntax like:

```yaml
security:
  - httpBearerToken: []
```

My proposal is that we do the following:

```yaml
...
  security:
    - $ref: '#/components/securitySchemes/httpBearerToken'
...  
components:
  securitySchemes:
    httpBearerToken:
      type: http
      scheme: bearer
```

The array of the current syntax is just for OAuth2 and OpenIDConnect schemes so I created a new property called `scopes` in the Security Scheme Object. These will be the mandatory scopes needed to connect or perform an operation. It offers less reusability in the `security` property but you can anyway create as many Security Scheme Objects inside components so, yeah, not an issue.

Reply:
I like it!

Reply:
Consider that in the security field we have 3 options:
- defining one security (as in your case)
- defining several security that must be met at once:
  
  ```yaml
  ...
  security: 
    - - $ref: '#/components/securitySchemes/1schema'
      - $ref: '#/components/securitySchemes/2schema'
  ```

- defining several security, for which only one security must be satisfied. 

  ```yaml
  ...
  security: 
    - $ref: '#/components/securitySchemes/1schema'
    - $ref: '#/components/securitySchemes/2schema'
  ```
  
So as you can see, we have problem with second case (several securities have to met at once). We have even issue for that https://github.com/asyncapi/spec/issues/828

Reply:
> The array of the current syntax is just for OAuth2 and OpenIDConnect schemes so I created a new property called scopes in the Security Scheme Object. These will be the mandatory scopes needed to connect or perform an operation. It offers less reusability in the security property but you can anyway create as many Security Scheme Objects inside components so, yeah, not an issue.

Why not to create security item shape as:

```yaml
security: 
  - scheme:
       $ref: '#/components/securitySchemes/1schema'
    scopes: [...]
```

?

Reply:
> Why not to create security item shape as:
> 
> ```yaml
> security: 
>   - scheme:
>        $ref: '#/components/securitySchemes/1schema'
>     scopes: [...]
> ```
> 
> ?

Why should we? :) The proposed syntax looks much cleaner IMHO. It only has the problem of meeting multiple security schemes at once but I'm not sure it should be a requirement. Left a comment at https://github.com/asyncapi/spec/issues/828#issuecomment-1280635558.

Reply:
I understand your concern but, honestly, I don't think most people would have dozens of OAuth schemes with different scopes, just probably a couple or three. In any case, you can still reuse it this way, right?

```yaml
...
  security:
    - $ref: '#/components/securitySchemes/oauth2UserRead'
...  
components:
  securitySchemes:
    oauth2base:
      type: oauth2
      description: 'blah blah'
    oauth2UserRead:
      $ref: '#/components/securitySchemes/oauth2base'
      scopes: ['user-read']
    oauth2UserWrite:
      $ref: '#/components/securitySchemes/oauth2base'
      scopes: ['user-write']
```

Or are we planning not to support merging an object when using `$ref` alongside other properties?

Reply:
> Or are we planning not to support merging an object when using $ref alongside other properties?

As I remember, we have issue for that (or mention about it in another issue, I don't know), but I'm not involving with it. cc @smoya

Reply:
> As I remember, we have issue for that (or mention about it in another issue, I don't know), but I'm not involving with it. cc @smoya

@magicmatatjahu you raised it as thoughts here: https://github.com/asyncapi/spec/issues/778#issuecomment-1117128502

I don't think it has been discussed anywhere else 🧐 

> Or are we planning not to support merging an object when using $ref alongside other properties?

It's not that we are planning not to support it, but we have not planned or had a discussion about whether that should be a feature I think 🤔

Reply:
>And tooling is already behaving this way, I think.

That is only because the tooling has bugs, it is not intended behavior... 🙂 

Adding this functionality will definitely complicate it from a spec perspective, but yea simplify it in this case 😆

Reply:
Bug or feature? 😛 To be honest, this is a thing I see many people doing/trying expecting it to work. But yeah, that's not a discussion for this thread sorry for the noise.

Reply:
I strongly encourage folks to file thoughts around potential reference merging in AsyncAPI as issues at the [unified referencing discussion repo](https://github.com/json-schema-org/referencing), where we can consider them alongside, for example, the limited merging of `summary` and `description` that OpenAPI 3.1 does outside of JSON Schema references.

This wouldn't mean AsyncAPI is blocked from taking other action, but unified referencing will only be unified if we know what's going on with each project.